app/api/events.py

```diff
@@ -155,6 +175,7 @@ def before_post(self, args, kwargs, data=None):
         user = User.query.filter_by(id=kwargs['user_id']).first()
         modules = Module.query.first()
         validate_event(user, modules, data)
+        validate_date(None, data)
```

Comment:
Why should date be verified before get?

Reply:
Sorry, I didn't get you. This is being called in before_post.
Am I missing something here?

Reply:
my mistake